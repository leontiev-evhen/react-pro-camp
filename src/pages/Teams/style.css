.section-team {
  position: relative;
  height: 220px;
  text-align: center;
}

.section-team small {
  position: absolute;
  bottom: -20px;
  left: 0;
  z-index: 1;
  display: block;
  width: 100%;
  margin: 0;
  padding: 0;
  color: rgba(0, 0, 0, .5);
  text-transform: uppercase;
  font-size: 10px;
  text-decoration: none;
  text-align: center;
  -webkit-transition: .3s ease-in-out;
  transition: .3s ease-in-out;
  opacity: 0;
}

.section-team figure:hover+small {
  bottom: 5px;
  opacity: 1;
}

.section-team-title {
  margin-bottom: 10px;
  text-transform: uppercase;
  font-size: 10px;
}

.section-team figure img {
  -webkit-transform: scale(1);
  transform: scale(1);
  -webkit-transition: .3s ease-in-out;
  transition: .3s ease-in-out;
}

.section-team figure:hover img {
  -webkit-transform: scale(1.1);
  transform: scale(1.1);
}

.section-team a {
  color: #000;
}

.section-team a:hover {
  text-decoration: none;
}