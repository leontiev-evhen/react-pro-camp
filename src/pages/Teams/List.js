import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Row, Col, Figure } from 'react-bootstrap';
import { ErrorBoundary } from '../../components';
import './style.css';

function List({ teams }) {
	let i,
		j,
		list = [],
		tempArray = [],
		chunk = 4;

	for (i = 0, j = teams.length; i < j; i += chunk) {
		tempArray = teams.slice(i, i + chunk);
		let items = [];

		tempArray.map(item => {
			return items.push(
				<ErrorBoundary key={item.team_id}>
					<Col md="3">
						<div className="section-team">
							<Link to={`/team/${item.team_id}`}>
								<div className="section-team-title">{item.name}</div>
								<Figure>
									<Figure.Image alt="logo" src={item.logo} />
								</Figure>
								<small>{item.venue_city}</small>
							</Link>
						</div>
					</Col>
				</ErrorBoundary>
			);
		});
		list.push(
			<div key={i} className="my-3">
				<Row>{items}</Row>
			</div>
		);
	}
	return <ul>{list}</ul>;
}

List.propTypes = {
	teams: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default List;
