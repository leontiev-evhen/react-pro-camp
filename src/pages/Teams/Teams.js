import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { requestTeams } from '../../actions';
import ContainerHOC from '../../hoc/ContainerHOC';
import List from './List';

const DivContainerHOC = ContainerHOC(List);

export class Teams extends Component {
	componentDidMount() {
		if (this.props.teams.length === 0) this.props.fetchTeams();
	}

	render() {
		const { teams, isFetching, error } = this.props;

		return (
			<DivContainerHOC teams={teams} isFetching={isFetching} error={error} />
		);
	}
}

Teams.propTypes = {
	teams: PropTypes.array.isRequired,
	isFetching: PropTypes.bool.isRequired,
	error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
	fetchTeams: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
	teams: state.teams.teams,
	isFetching: state.teams.isFetching,
	error: state.teams.error,
});

const mapDispatchToProps = dispatch => ({
	fetchTeams: () => {
		dispatch(requestTeams());
	},
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Teams);
