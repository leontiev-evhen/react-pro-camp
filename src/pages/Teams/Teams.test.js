import React from 'react';
import { shallow } from 'enzyme';
import { Teams } from './Teams';

describe('Teams component', () => {
	const props = {
		teams: [],
		isFetching: false,
		error: '',
		fetchTeams: jest.fn(),
	};

	const Component = shallow(<Teams {...props} />);

	it('componentDidMount should be called', () => {
		jest.spyOn(Teams.prototype, 'componentDidMount');
		shallow(<Teams {...props} />);
		expect(Teams.prototype.componentDidMount).toHaveBeenCalled();
	});

	it('Teams component render correctly', () => {
		expect(Component).toMatchSnapshot();
	});

	describe('Teams component already has teams props', () => {
		const nextProps = {
			teams: [
				{
					name: 'team1',
				},
				{
					name: 'team2',
				},
				{
					name: 'team3',
				},
			],
			isFetching: false,
			error: '',
			fetchTeams: jest.fn(),
		};
		shallow(<Teams {...nextProps} />);
		it('Function fetchTeams should not be called', () => {
			expect(nextProps.fetchTeams).toHaveBeenCalledTimes(0);
		});
	});
});
