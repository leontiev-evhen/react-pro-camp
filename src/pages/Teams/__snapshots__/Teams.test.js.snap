// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams component Teams component render correctly 1`] = `
<containerHOC
  error=""
  isFetching={false}
  teams={Array []}
/>
`;
